charts/geoserver/templates/deployment.yaml

```diff
@@ -34,36 +34,47 @@ spec:
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           env:
+            - name: INSTALL_EXTENSIONS
+              value: {{ .Values.extension.install | quote }}
+            - name: STABLE_EXTENSIONS
+              value: {{ .Values.extension.stableExtension | quote }}
+            - name: CORS_ENABLED
+              value: {{ .Values.cors.enabled | quote }}
+            - name: SKIP_DEMO_DATA
+              values: {{ .Values.demoData.skip | quote }}
+            - name: GEOSERVER_CSRF_WHITELIST
+              values: {{ .Values.geoserver.whitelist | quote }}
             - name: GEOSERVER_ADMIN_USER
-              valueFrom:
-                secretKeyRef:
-                  name: {{ include "geoserver.fullname" . }}
-                  key: GEOSERVER_ADMIN_USER
+              value: admin
             - name: GEOSERVER_ADMIN_PASSWORD
+              {{- if .Values.auth.existingSecret }}
               valueFrom:
                 secretKeyRef:
-                  name: {{ include "geoserver.fullname" . }}
-                  key: GEOSERVER_ADMIN_PASSWORD
+                  name: {{ .Values.auth.existingSecret }}
+                  key: {{ .Values.auth.passwordKey }}
+              {{- else }}
+              value: {{ .Values.auth.password }}
```

Comment:
no matter what we should use a secret for the password. In this case we have to create one.

Reply:
changed it to only use the secret